src/scmdata/run.py

```diff
@@ -1500,19 +1500,51 @@ def convert_unit(
         :obj:`ScmRun`
             If :obj:`inplace` is not ``False``, a new :class:`ScmRun` instance
             with the converted units.
+
+        Notes
+        -----
+        If ``context`` is not ``None``, then the context used for the conversion will
+        be checked against any existing metadata and, if the conversion is valid,
+        stored in the output's metadata.
+
+        Raises
+        ------
+        ValueError
+            ``"unit_context"`` is already included in ``self``'s :meth:`meta_attributes`
+            and it does not match ``context`` for the variables to be converted.
         """
         # pylint: disable=protected-access
         if inplace:
             ret = self
         else:
             ret = self.copy()
 
-        if "unit_context" not in ret.meta_attributes:
-            ret["unit_context"] = None
-
         to_convert = ret.filter(**kwargs)
         to_not_convert = ret.filter(**kwargs, keep=False, log_if_empty=False)
 
+        if "unit_context" in to_convert.meta_attributes:
+            unit_context = to_convert.get_unique_meta("unit_context")
+            # check if contexts don't match, unless the context is nan
+            non_matching_contexts = len(unit_context) > 1 or unit_context[0] != context
+            if isinstance(unit_context[0], float):
+                non_matching_contexts &= not np.isnan(unit_context[0])
+
+            if non_matching_contexts:
+                raise ValueError(
+                    "Existing unit conversion context(s), `{}`, doesn't match input "
+                    "context, `{}`, drop `unit_context` metadata before doing "
+                    "conversion".format(unit_context, context)
+                )
+
+            to_convert["unit_context"] = context
+
+        elif context is not None:
+            to_convert["unit_context"] = context
+
+        if "unit_context" not in to_not_convert.meta_attributes and context is not None:
+            # what to do here?
```

Comment:
It's a small enough use case that all unit contexts have to be specified

Reply:
```suggestion

```

I think the comment can simply be deleted. For now I've just been setting `np.nan`, seems to work ok in the tests.